### PR TITLE
Filters with api

### PR DESCRIPTION
**Pull request by StRhys**
This is to add api calls for the filters.

to test
play around with filters, subtopics should only appear when a topic has been picked.
Give the page a few refreshes, see if filters persist.

### CODE REVIEW

File: app/data/topicItems.json [hunk 1]
```diff
@@ -2,47 +2,47 @@
   "topics": [
```
Comment (mikeAdamss) [inline app/data/topicItems.json:2]: so can we lose this file or is it referenced elsewhere?

**Review by mikeAdamss — COMMENTED** [review]
So the selection mechanics work grand but the datasets filtering looks like its off, as an example:

- I get two datasets if I filter by parent topic of. economy.
- I get 0 datasets if I filter by climate change

File: app/data/topicItems.json (hunk 1 above)
Comment (StRhys) [reply app/data/topicItems.json:2]: Final place is in the header, will make sure to remove it once we get to that.

**Comment by StRhys** [comment]
There's a slight mismatch between what the api is returning for topics and datasets.
Topics returns `climate-change` and datasets is returning `climatechange`, so throwing the filters off.

**Review by mikeAdamss — APPROVED** [review]
LGTM

**Pull request merged**